app/payment-link-v2/reference-confirm/reference-confirm.controller.js

```diff
@@ -21,20 +45,52 @@ function getPage (req, res, next) {
     paymentReferenceHint: product.reference_hint
   }
 
-  data.referencePageUrl = replaceParamsInPath(paymentLinksV2.reference, product.externalId)
+  const sessionReferenceNumber = paymentLinkSession.getReference(req, product.externalId)
+
+  if (sessionReferenceNumber) {
+    data.reference = sessionReferenceNumber
+  }
+
+  const backLinkHref = replaceParamsInPath(paths.paymentLinksV2.reference, product.externalId)
+  data.backLinkHref = backLinkHref
 
-  const confirmAndContinueUrlPath = product.price ? paymentLinksV2.confirm : paymentLinksV2.amount
-  data.confirmAndContinuePageUrl = replaceParamsInPath(confirmAndContinueUrlPath, product.externalId)
+  return response(req, res, 'reference-confirm/reference-confirm', data)
+}
+
+function postPage (req, res, next) {
+  const confirmReference = lodash.get(req.body, CONFIRM_REFERENCE, '')
+  const product = req.product
+  const { change } = req.query || {}
```

Comment:
Good spot that this should direct to the confirm page if the reference is being edited.

I think we'll need to pass on the `?change=true` query parameter from the reference page on the redirect though.

Reply:
Ah - good point.  I will look into that, thanks

Reply:
Just updated it - please take a look